core/vmpu/src/vmpu.c

```diff
@@ -413,6 +419,33 @@ static void vmpu_enumerate_boxes(void)
         /* Select the pointer to the (permuted) box configuration table. */
         int index = box_order[box_id];
         UvisorBoxConfig const * box_cfgtbl = ((UvisorBoxConfig const * *) __uvisor_config.cfgtbl_ptr_start)[index];
+        TUvisorDebugDriver const * const debug_driver = __uvisor_config.debug_driver;
+
+        /* If a debug driver was created by the application and it is associated to this box_id,
+         * initialize g_debug_box accordingly.
+         * Note: The verification that no more than one box was created and been associated with the debug driver
+         * is done statically on debug driver creation time [in build time] */
+        if ((debug_driver != NULL) && (debug_driver->box_cfg_ptr == box_cfgtbl)) {
```

Comment:
`debug_driver` should never be `NULL`. Consider changing to an `assert` or `HALT_ERROR(SANITY_CHECKFAILED`. It's a programmer error if `debug_driver` is `NULL`.